be/src/olap/tablet_meta.cpp

```diff
@@ -17,968 +17,197 @@
-    }
-
-    VLOG(10) << "calculated shortest path. "
-             << "version=" << target_version.first << "-" << target_version.second
-             << "path=" << shortest_path_for_debug.str();
-
-    return OLAP_SUCCESS;
+NewStatus TabletMeta::save_meta() {
+    std::lock_guard<std::mutex> lock(_mutex);
+    return save_meta_unlock();
 }
 
-const PDelta* TabletMeta::get_lastest_delta_version() const {
-    if (delta_size() == 0) {
-        return nullptr;
-    }
-
-    const PDelta* max_delta = nullptr;
-    for (int i = delta_size() - 1; i >= 0; --i) {
-        if (delta(i).start_version() == delta(i).end_version()) {
-            if (max_delta == nullptr) {
-                max_delta = &delta(i);
-            } else if (delta(i).start_version() > max_delta->start_version()) {
-                max_delta = &delta(i);
-            }
-        }
+NewStatus TabletMeta::save_meta_unlock() {
+    string meta_binary;
+    serialize_unlock(meta_binary);
+    NewStatus status = TabletMetaManager::save(_data_dir, _tablet_id, _schema_hash, meta_binary);
+    if (!status.ok()) {
+       LOG(WARNING) << "fail to save tablet_meta. status=" << status.to_string()
+                    << ", tablet_id=" << _tablet_id
+                    << ", schema_hash=" << _schema_hash;
     }
-    if (max_delta != nullptr) {
-        LOG(INFO) << "max_delta:" << max_delta->start_version() << ","
-            << max_delta->end_version();
-    }
-    return max_delta;
+    return status;
 }
 
-const PDelta* TabletMeta::get_lastest_version() const {
-    if (delta_size() == 0) {
-        return nullptr;
-    }
-
-    const PDelta* max_delta = nullptr;
-    for (int i = delta_size() - 1; i >= 0; --i) {
-        if (max_delta == nullptr) {
-            max_delta = &delta(i);
-        } else if (delta(i).end_version() > max_delta->end_version()) {
-            max_delta = &delta(i);
-        } else if (delta(i).end_version() == max_delta->end_version()
-                   && delta(i).start_version() == delta(i).end_version()) {
-            max_delta = &delta(i);
-        }
-    }
-    return max_delta;
+NewStatus TabletMeta::to_meta_pb(TabletMetaPB* tablet_meta_pb) {
+    std::lock_guard<std::mutex> lock(_mutex);
+    return to_meta_pb_unlock(tablet_meta_pb);
 }
 
-Version TabletMeta::get_latest_version() const {
-    auto delta = get_lastest_version();
-    return {delta->start_version(), delta->end_version()};
-}
+NewStatus TabletMeta::to_meta_pb_unlock(TabletMetaPB* tablet_meta_pb) {
+    tablet_meta_pb->set_table_id(_table_id);
+    tablet_meta_pb->set_partition_id(_partition_id);
+    tablet_meta_pb->set_table_id(_tablet_id);
+    tablet_meta_pb->set_schema_hash(_schema_hash);
+    tablet_meta_pb->set_shard_id(_shard_id);
 
-const PDelta* TabletMeta::get_delta(int index) const {
-    if (delta_size() == 0) {
-        return nullptr;
+    tablet_meta_pb->set_tablet_name(_tablet_name);
```

Comment:
what is tablet name?

Reply:
table_name may not stored in PB, otherwise cannot to be renamed.